packages/reapit-connect-management/src/components/ui/offices/edit-office-group.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react'
+import useSWR from 'swr'
+import { FormFieldInfo } from '@reapit/utils'
+import { OfficeGroupModel } from '@reapit/foundations-ts-definitions'
+import {
+  Button,
+  Section,
+  ModalV2,
+  H5,
+  Formik,
+  Form,
+  Input,
+  DropdownSelect,
+  Loader,
+  notification,
+} from '@reapit/elements'
+import { URLS } from '../../../constants/api'
+import { updateOfficeGroup } from '../../../services/office'
+import { errorMessages } from '../../../constants/errorMessages'
+import { prepareOfficeOptions } from '../../../utils/prepareOptions'
+
+export interface UpdateOfficeGroupModalProps {
+  editingGroup: OfficeGroupModel | undefined
+  setEditingGroup: React.Dispatch<React.SetStateAction<OfficeGroupModel | undefined>>
+  orgId: string
+  onRefetchData: Function
+}
+
+export interface UpdateOfficeGroupModel {
+  name: string
+  officeIds: string[]
+}
+
+export type FieldType = 'name' | 'officeIds'
+
+export const formFields: Record<FieldType, FormFieldInfo> = {
+  name: {
+    name: 'name',
+    label: 'Name',
+  },
+  officeIds: {
+    name: 'officeIds',
+    label: 'Offices',
+  },
+}
+
+export const UpdateOfficeGroupModal: React.FC<UpdateOfficeGroupModalProps> = ({
+  editingGroup,
+  setEditingGroup,
+  orgId,
+  onRefetchData,
+}) => {
+  const handleOnClose = () => setEditingGroup(undefined)
+  const { name, officeIds } = formFields
+
+  const { data }: any = useSWR(`${URLS.OFFICES}`)
+  if (!data) return <Loader />
+  const { _embedded: listOffice } = data
+  const officeOptions = prepareOfficeOptions(listOffice)
+
+  const onSubmit = async (params: UpdateOfficeGroupModel) => {
```

Comment:
Can you break this onSubmit logic out of the main body function so it can be tested - there is quite a lot of logic in here!